### ISSUE TITLE
Crash in logic that checks return type

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6632]
Stack trace:

```python
TypeError: Cannot read properties of undefined (reading 'nodeType')
at nodeType src/analyzer/analyzerNodeInfo.ts:148:16
at getFileInfo src/analyzer/typeEvaluator.ts:3083:42
at callback src/common/timing.ts:40:19
at addDiagnostic src/analyzer/checker.ts:1010:32
at visitReturn src/analyzer/parseTreeWalker.ts:526:28
at visit src/analyzer/parseTreeWalker.ts:933:20
at visitNode src/analyzer/parseTreeWalker.ts:915:36
at walk src/analyzer/checker.ts:267:18
at declaration src/pytest/pyTestDocumentTransformer.ts:180:12
at func (nonymous>)
at forEach src/analyzer/parseTreeWalker.ts:922:14
at walkMultiple src/analyzer/parseTreeWalker.ts:917:17
at walk src/analyzer/checker.ts:267:18
at walk src/analyzer/checker.ts:2695:17
at _walkStatementsAndReportUnreachable src/analyzer/checker.ts:276:13
at visitSuite src/analyzer/parseTreeWalker.ts:544:28
at visit src/analyzer/parseTreeWalker.ts:933:20
at visitNode src/analyzer/parseTreeWalker.ts:915:36
at walk src/analyzer/checker.ts:267:18
at declaration src/pytest/pyTestDocumentTransformer.ts:180:12
at func (nonymous>)
at forEach src/analyzer/parseTreeWalker.ts:922:14
at walkMultiple src/analyzer/parseTreeWalker.ts:917:17
at walk src/analyzer/checker.ts:267:18
at walk src/analyzer/checker.ts:2695:17
at _walkStatementsAndReportUnreachable src/analyzer/checker.ts:276:13
at visitSuite src/analyzer/parseTreeWalker.ts:544:28
at visit src/analyzer/parseTreeWalker.ts:933:20
at visitNode src/analyzer/parseTreeWalker.ts:915:36
at walk src/analyzer/checker.ts:267:18
at walk src/analyzer/checker.ts:643:17
at visitFunction src/analyzer/parseTreeWalker.ts:442:28
```

This is copied from https://github.com/microsoft/pyright/issues/6391.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
